app/server/pkg/community/google/analytics.go

```diff
@@ -143,5 +148,42 @@ func getGraphData() error {
 		dailyExperimentData[i][0] = val[0][:4] + "-" + val[0][4:6] + "-" + val[0][6:]
 	}
 	GAResponseJSONObject.DailyExperimentData = dailyExperimentData
+
+	response, err = svc.Data.Ga.Get(viewID, startDate, endDate, "ga:totalEvents").Dimensions("ga:yearMonth").Filters(filters).Filters("ga:eventLabel==Chaos-Operator").MaxResults(999999999).Do()
+	if err != nil {
+		return fmt.Errorf("Error while getting response, err: %s", err)
+	}
+	monthlyOperatorData := response.Rows
+	for i, val := range monthlyOperatorData {
```

Comment:
What is this block doing?

Reply:
This is returning the (month of year) wise data after preprocessing for all the edge cases in the google analytics server itself, but it does not add the last date of each month that is necessary for the time series on litmus portal, so i am adding it manually.

Reply:
Put it in a function and call it instead of duplicating so much code. Also the logic for Feb is wrong, we have to handle leap years

Reply:
I will be adding the leap year check asap, for now i have made it modular.

Reply:
The leap year check has been added too, in the new PR #200